File: C#/151. Reverse Words in a String.cs
public class Solution {
    public string ReverseWords(string s) {
        string ans = "";
        string w = "";
        int i = 0;
        while(i < s.Length){
            while(i < s.Length && s[i] != ' '){
                w += s[i];
                i++;
            }
            if (w.Length != 0) {
                if (ans.Length != 0) {
                    ans = w + " " + ans;
                }
                else {
                    ans = w;
                }
                w = "";
            }
            i++;
        }
        return ans;
    }
}

File: Java/151. Reverse Words in a String.java
class Solution {
    public String reverseWords(String s) {
        String ans = "";
        String w = "";
        int i = 0;
        while(i < s.length()){
            while(i < s.length() && s.charAt(i) != ' '){
                w += s.charAt(i);
                i++;
            }
            if (w.length() != 0) {
                if (ans.length() != 0) {
                    ans = w + " " + ans;
                }
                else {
                    ans = w;
                }
                w = "";
            }
            i++;
        }
        return ans;
    }
}
